types/block.go

```diff
@@ -90,11 +90,11 @@ func (b *Block) ValidateBasic() error {
 	// NOTE: b.Evidence.Evidence may be nil, but we're just looping.
 	for i, ev := range b.Evidence.Evidence {
 		switch ev.(type) {
-		case *ConflictingHeadersEvidence, ConflictingHeadersEvidence:
+		case *ConflictingHeadersEvidence:
 			// ConflictingHeadersEvidence must be broken up in pieces and never
 			// committed as a single piece.
 			return fmt.Errorf("found ConflictingHeadersEvidence (#%d)", i)
-		case *PotentialAmnesiaEvidence, PotentialAmnesiaEvidence:
```

Comment:
I'd prefer for these to stay just in case

Reply:
I was getting these errors: 
```
types/block.go:93:3: impossible type switch case: ev (type Evidence) cannot have dynamic type ConflictingHeadersEvidence (Address method has pointer receiver)
types/block.go:97:3: impossible type switch case: ev (type Evidence) cannot have dynamic type PotentialAmnesiaEvidence (Address method has pointer receiver)
```